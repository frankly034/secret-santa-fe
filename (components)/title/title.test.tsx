import { render } from "@testing-library/react";

import Title from ".";

describe("Title - Component", () => {
  it("renders title component", () => {
    const { container } = render(<Title>{"Gift a friend this holiday"}</Title>);
    expect(container).toMatchSnapshot();
  });

  it("renders single string child", () => {
    const { getByRole } = render(<Title>{"Gift"}</Title>);
    expect(getByRole("heading")).toBeTruthy;
  });

  it("should render last word with lastWordColor", () => {
    const { getByText } = render(
      <Title lastWordColor="#ff0000">Gift a friend this holiday</Title>
    );
    expect(getByText("holiday").style.color).toEqual("rgb(255, 0, 0)");
  });

  it("should render preceeding texts and last word with appropriate colors", () => {
    const { getByText } = render(
      <Title preceedingWordsColor="#00ff00" lastWordColor="#ff0000">
        Gift a friend this holiday
      </Title>
    );
    expect(getByText("Gift a friend this").style.color).toEqual(
      "rgb(0, 255, 0)"
    );
    expect(getByText("holiday").style.color).toEqual("rgb(255, 0, 0)");
  });
});
