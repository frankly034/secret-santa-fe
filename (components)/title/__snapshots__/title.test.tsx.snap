// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title - Component renders title component 1`] = `
<div>
  <h1
    class="ant-typography title css-dev-only-do-not-override-15rg2km"
  >
    Gift a friend this holiday
  </h1>
</div>
`;
